File: frontend/src/models/IncomeType.ts
export enum IncomeTypeEnum {
  Manual = 0,
  Hourly = 1
}

export interface IncomeType {
  id: number
  name: string
  type: IncomeTypeEnum
}

export interface IncomeTypeFormData {
  name: string
  type: IncomeTypeEnum
}

File: backend/Models/IncomeType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MonthBalance.Models;

public class IncomeType
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [Required]
    public IncomeTypeEnum Type { get; set; } = IncomeTypeEnum.Manual;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Income> Incomes { get; set; } = new List<Income>();
}
